packages/rollup-contracts/contracts/CanonicalTransactionChain.sol

```diff
@@ -5,12 +5,14 @@ pragma experimental ABIEncoderV2;
 import {DataTypes as dt} from "./DataTypes.sol";
 import {RollupMerkleUtils} from "./RollupMerkleUtils.sol";
 import {L1ToL2TransactionQueue} from "./L1ToL2TransactionQueue.sol";
+import {SafetyTransactionQueue} from "./SafetyTransactionQueue.sol";
 
 contract CanonicalTransactionChain {
   address public sequencer;
   uint public forceInclusionPeriod;
   RollupMerkleUtils public merkleUtils;
   L1ToL2TransactionQueue public l1ToL2Queue;
+  SafetyTransactionQueue public safetyQueue;
```

Comment:
I don't ever see this queue or the `l1ToL2Queue` get added to. Is that logic to be written in the future, or am I missing something?

Reply:
That logic is in the RollupQueue contracts, which both queues extend: https://github.com/ethereum-optimism/optimism-monorepo/blob/576c035586f50313bbd6a898ad307d6b2487faaf/packages/rollup-contracts/contracts/RollupQueue.sol#L45-L52